src/main/java/com/softserve/service/impl/GroupServiceImpl.java

```diff
@@ -127,23 +128,20 @@ public Group updateGroupOrder(Group group, Long afterId) {
         if (!groupRepository.isExistsById(group.getId())) {
             throw new EntityNotFoundException(Group.class, "id", group.getId().toString());
         }
+        Integer maxOrder = groupRepository.getMaxSortingOrder().orElse(0);
         if (afterId != null) {
-            Double previousPosition = getSortingOrderById(afterId);
-            Double nextPosition = groupRepository.getNextPosition(previousPosition)
-                    .orElse(previousPosition+2);
-            Double newPosition = (previousPosition+nextPosition)/2;
-            if (newPosition - 1 < 0.01) {
-                newPosition = groupRepository.getMaxSortingOrder().orElse(0.0)+1;
-            }
-            group.setSortingOrder(newPosition);
+            Integer lowerBound = getSortingOrderById(afterId)+1;
+            Integer upperBound = Optional.ofNullable(group.getSortingOrder()).orElse(maxOrder+1)+1;
```

Comment:
Didn't get why do you need 'upperBound'

Reply:
It is used for update, to change sort orders only in specific interval

Reply:
> It is used for update, to change sort orders only in specific interval

Reply:
It is used for update, to change sort orders only in specific interval